test/e2e/template/helmfile/snapshot_test.go

```diff
@@ -200,6 +213,15 @@ func TestHelmfileTemplateWithBuildCommand(t *testing.T) {
 				}
 			}
 
+			tmpDir := t.TempDir()
+			// HELM_CACHE_HOME contains downloaded chart archives
+			helmCacheHome := filepath.Join(tmpDir, "helm_cache")
```

Comment:
I had to give clean HELM_CACHE_HOME and HELM_CONFIG_HOME to every snapshot test case to make it not affected by the previous tests.